File: CircleDI.Tests/GeneratorTests/AcceptAllTests.py
###########################################
#
# goes throgh the folder and subfolders
#   if file name is "*.received.txt"
#     rename file to "*.verified.txt" (and overwrite existing file if any)
#
###########################################


import os
import subprocess

def accept_folder(directory: str):
    for (dirpath, dirnames, filenames) in os.walk(directory):
        for filename in filenames:
            if filename.endswith(".received.txt"):
                basename = filename[:-13]
                os.system(f"move /y {dirpath}\{basename}.received.txt {dirpath}\{basename}.verified.txt")
                print(f"accepted {basename}\n")


accept_folder(os.path.dirname(__file__));
accept_folder(os.path.join(os.path.dirname(__file__), "..", "BlazorTests"));
accept_folder(os.path.join(os.path.dirname(__file__), "..", "MinimalAPITests"));

File: CircleDI.Tests/AcceptAllTests/AcceptAllTests.cs
#!/usr/bin/env dotnet

/**
 * goes throgh the folder and subfolders
 *   if file name is "*.received.txt"
 *     rename file to "*.verified.txt" (and overwrite existing file if any)
 **/

string currentDirectory = Directory.GetCurrentDirectory();
AcceptFolder(Path.Join(currentDirectory, "..", "GeneratorTests"));
AcceptFolder(Path.Join(currentDirectory, "..", "BlazorTests"));
AcceptFolder(Path.Join(currentDirectory, "..", "MinimalAPITests"));

void AcceptFolder(string directory) {
    foreach (string fileName in Directory.GetFiles(directory, "*", SearchOption.AllDirectories))
        if (fileName.EndsWith(".received.txt")) {
            string baseName = fileName[..^".received.txt".Length];
            File.Move(fileName, $"{baseName}.verified.txt", overwrite: true);
            Console.WriteLine($"accepted {Path.GetFileName(baseName)}");
        }
}
